### ISSUE TITLE
PSR2.Methods.FunctionCallSignature.SpaceBeforeCloseBracket false positive for heredoc/nowdoc

### ISSUE THREAD

**Issue by mabar** [squizlabs/PHP_CodeSniffer#3706]
**Describe the bug**

`PSR2.Methods.FunctionCallSignature.SpaceBeforeCloseBracket` forbids space before close bracket, but it should not be reported if last value in call is heredoc/nowdoc

**Code sample**

This is okay

```php
function test()
{
	foo(
		'bar',
		'baz',
	);

	foo(
		<<<'MSG'
Long
Message
MSG,
	);
}
```

This is reported by rule

```php
function test(): void
{
	foo(<<<'MSG'
Long
Message
MSG,
	);
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset>
  <rule ref="PSR2.Methods.FunctionCallSignature.SpaceBeforeCloseBracket"/>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed

```
Expected 0 spaces before closing parenthesis; newline found (PSR2.Methods.FunctionCallSignature.SpaceBeforeCloseBracket)
```

**Expected behavior**
No error.

Interesting is that similar code without heredoc/nowdoc does not report problematic check, but instead reports `PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket`

```php
function test(): void
{
	foo('bar',
		'baz',
	);
}
```

Reporting fixable `ContentAfterOpenBracket` would also work (and should be probably fixed too), because in case `ContentAfterOpenBracket` is fixed, then `SpaceBeforeCloseBracket` is no longer reported

**Versions:**
 - OS: Windows 10
 - PHP: 8.1.9
 - PHPCS: 3.7.1
 - Standard: orisai/coding-standard-php (but mostly irrelevant for the issue)
